### PR TITLE
[boost] rename boost helpers

### PR DESCRIPTION
**Pull request by yurybura**
This PR renames vcpkg helpers for install Boost libraries:
 boost-vcpkg-helpers -> vcpkg-boost-copy
 boost-modular-build-helper -> vcpkg-boost-build
 boost-uninstall -> vcpkg-boost-uninstall

### CODE REVIEW

**Comment by JackBoosY** [comment]
Does this change have any effect other than renaming? If not, I think we no need to do that.

**Comment by yurybura** [comment]
> Does this change have any effect other than renaming? If not, I think we no need to do that.

This PR contains only changes related to the refactoring. Rename is the biggest one.

**Comment by dg0yt** [comment]
Caution with changing to host ports. A plain `vcpkg remove --recurse boost-uninstall` can only remove the current triplet's dependencies. IMO `boost-uninstall` should remain in the target triplet.

**Comment by strega-nil-ms** [comment]
What is the purpose for renaming these? There needs to be a significant bar for renaming something like this, given the massive DAG of changes that is required.

**Comment by yurybura** [comment]
> What is the purpose for renaming these? There needs to be a significant bar for renaming something like this, given the massive DAG of changes that is required.

Purposes: 
1) Separate the Boost libraries (boost-<libname>) from vcpkg's internal ports.
2) Make all build helpers as host ports.

Almost all changes were made by the ports generation script.

**Comment by yurybura** [comment]
@PhoebeHui Please have a look this PR. I hope your team can decide ASAP. New Boost 1.79 is coming soon.

**Comment by dg0yt** [comment]
IIUC this PR implicitly turns `boost-uninstall` into a host dependency. This would break the usual `vcpkg remove boost-uninstall` pattern. At least this port must stay with the target triplet IMO.

**Comment by PhoebeHui** [comment]
@yurybura, sorry for the late response! I will bring this to our meeting and get back to you ASAP.

File: ports/boost-uninstall/vcpkg.json [hunk 1]
```diff
@@ -1,5 +0,0 @@
-{
```
Comment (strega-nil-ms) [inline ports/boost-uninstall/vcpkg.json:1]: this port still needs to exist, even as an empty port.

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]

**Comment by strega-nil-ms** [comment]
I'll bring this up at the thursday meeting

**Comment by strega-nil-ms** [comment]
Unfortunately, we are not interested in taking this change; for one, the benefit of them is not worth the churn, and in the future we would need to keep the old names around anyways. If we were introducing these for the first time, these names would be great, but we aren't - we need to keep compatibility with old names.

**Pull request closed without merging**